src/generic/Matrix.jl

```diff
@@ -502,19 +502,145 @@ function setindex!(a::AbstractAlgebra.MatElem{T}, b::AbstractAlgebra.MatElem{T},
     end
 end
 
-setindex!(a::AbstractAlgebra.MatElem{T}, b::AbstractAlgebra.MatElem{T}, r::UnitRange{Int}, ::Colon) where T = setindex!(a, b, r, 1:ncols(a))
+function setindex!(a::MatrixElem{T}, b::Matrix{<:RingElement}, r::UnitRange{Int}, c::UnitRange{Int}) where T
```

Comment:
Can't this method be merged with the previous one by writing `b::Union{MatrixElem, Matrix}` ?